frontend/src/metabase/lib/redux/hooks.unit.spec.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import type { State } from "metabase-types/store";
+import { createMockUser } from "metabase-types/api/mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+
+const DEFAULT_USER = createMockUser({ email: undefined });
+const TEST_EMAIL = "test_email@metabase.test";
+
+describe("useSelector", () => {
```

Comment:
I think we should not test the external library.

With the RTK the right approach to type it would be like [this](https://redux-toolkit.js.org/usage/usage-with-typescript#getting-the-dispatch-type) but it will require actually doing the full migration https://github.com/metabase/metabase/issues/29637 so any intermediate solution that improves typings for now would work

Reply:
@alxnddr  So the approach provided by RTK is one valid approach to typing. Keep in mind RTK is not doing anything special itself for thunks, it just includes the `redux-thunk` library with it. We don't have to do the full migration described in https://github.com/metabase/metabase/issues/29637 to get typesafety for `useSelector` and `useDispatch` (which is the lone goal of this PR), we would just need to add the two-line re-export.

The downside to this approach is that we have to re-export `useSelector` and `useDispatch`, and anyone who wants to use them will have to be sure to import them from `metabase/lib/redux` and not from `react-redux`. This was originally the approach that I had in an earlier version of the PR, but @kulyk pointed out the disadvantages to exporting, and suggested using `/// <reference types="redux-thunk/extend-redux" />`, which gives us typesafety on the hooks imported directly from the original `react-redux` library. This is what we are currently doing in this PR.

As I described in [this comment](https://github.com/metabase/metabase/pull/29530#issuecomment-1499237304), we are currently blocked because our type check CI job is failing due to errors in the `redux-thunk` library, likely because we used the type reference.

**We have two possible paths forward:**

A) Go back to re-exporting `useSelector` and `useDispatch`. This will pass CI (it did before) and we would be able to merge this change soon™️ (e.g. end of this week or early next).

B) Fix our CI job so it does not type check external libraries that we have a type reference for. I have no idea when I or anyone else would be able to do this.

**Given that approach A is known to work, and that it is uncertain when or even if we can try approach B, I'm inclined to go with approach A.**

It is a little unfortunate that we do have to go with the suboptimal approach of re-exporting these hooks ourselves, I think it's better to have a slightly less optimal solution that we can start using now, rather than delaying this from merging and making it contingent on solving a problem that will take an uncertain amount of time to fix. If we really do want to import directly from `react-redux`, we can figure out how to do that later and refactor. For now I think re-exporting the hooks is a good enough stopgap solution.

@alxnddr @kulyk Let me know if you guys are aligned with this and if so I will revert to the previous version of the PR and merge.

Reply:
@EmmadUsmani I agree with you on that. When we migrate to rtk fully, we will still have re-exporting